### PR TITLE
Sync: Usage Pixels and Monitoring

### PR DESCRIPTION
**Pull request by malmstein**
Task/Issue URL: https://app.asana.com/0/488551667048375/1206193954622385/f

### Description
This PR adds the usage pixels for Sync

### Steps to test this PR

_Signup Direct_
Fired when user sets up a sync account directly.
- [x] Create a new sync account
- [x] Verify “m.sync.signup.direct” pixel is sent

_Signup Connect_
Fired when user sets up a sync account from connect flow. You’ll need two devices for this one.
- [x] Device A - Open Sync Setup screen
- [x] Tap on Sync with another device
- [x] Device B- Open Sync Setup screen
- [x] Tap on Sync with another device
- [x] Device A - Read QR code present in Device B
- [x] Verify “m.sync.signup.connect” pixel is sent in Device B
- [x] Verify “m.sync.login” pixel is sent in Device A

_Login - Entering Code_
Fired when adding new device to existing account. You’ll need two devices for this one.
- [x] Device A - Create a new sync account
- [x] Copy the recovery code
- [x] Device B- Open Sync Setup screen
- [x] Tap on Sync with another device
- [x] Tap on Manually Enter Code
- [x] Paste code from Device A
- [x] Verify “m.sync.login” pixel is sent in Device B

_Login - Scan QR_
Fired when adding new device to existing account. You’ll need two devices for this one.
- [x] Device A - Create a new sync account
- [x] Navigate to Sync Settings
- [x] Device B- Open Sync Setup screen
- [x] Tap on Sync with another device
- [x] Scan QR Code from Device A
- [x] Verify “m.sync.login” pixel is sent in Device B

_Login - Recover Account with Code_
Fired when adding new device to existing account. You’ll need two devices for this one.
- [x] Device A - Create a new sync account
- [x] Copy the recovery code
- [x] Device B- Open Sync Setup screen
- [x] Tap on Recover Synced Data
- [x] Tap on Manually Enter Code
- [x] Paste code from Device A
- [x] Verify “m.sync.login” pixel is sent in Device B

_Login - Recover Account with QR_
Fired when adding new device to existing account. You’ll need two devices for this one.
- [x] Device A - Create a new sync account
- [x] Navigate to Sync Settings
- [x] Device B- Open Sync Setup screen
- [x] Tap on Recover Synced Data
- [x] Scan QR Code from Device A
- [x] Verify “m.sync.login” pixel is sent in Device B

_Daily pixel_
Fired for first sync on a given day.
- [x] Create a new sync account
- [x] Verify “m.sync.daily” pixel is sent
- [x] Close app
- [x] Open app again
- [x] Verify “m.sync.daily” pixel is NOT sent

### CODE REVIEW

**Comment by malmstein** [comment]
Current dependencies on/for this PR:
* `develop`
  * **PR #4018** <a href="https://app.graphite.dev/github/pr/duckduckgo/Android/4018?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/duckduckgo/Android/4018?utm_source=stack-comment).

File: sync/sync-impl/src/main/java/com/duckduckgo/sync/impl/SyncAccountRepository.kt [hunk 1]
```diff
@@ -25,6 +25,7 @@ import com.duckduckgo.sync.api.engine.SyncEngine.SyncTrigger.ACCOUNT_CREATION
 import com.duckduckgo.sync.api.engine.SyncEngine.SyncTrigger.ACCOUNT_LOGIN
 import com.duckduckgo.sync.crypto.*
 import com.duckduckgo.sync.impl.Result.Error
+import com.duckduckgo.sync.impl.Result.Success
```
Comment (joshliebe) [inline sync/sync-impl/src/main/java/com/duckduckgo/sync/impl/SyncAccountRepository.kt:28]: A few unused imports in this file

**Review by joshliebe — APPROVED** [review]
Just some unused imports, otherwise LGTM and works as expected 🚀

**Comment by malmstein** [comment]
Thanks @joshliebe !

**Pull request merged**